### PR TITLE
feat: common names auto-generation

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Added a utility to generate integration common names automatically.
Also, relocated a handful of integration-specific utility methods to save bundle size.

A clone of https://github.com/rudderlabs/rudder-sdk-js/pull/685.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/774)

### CODE REVIEW

**Review by MoumitaM — CHANGES_REQUESTED** [review]

File: src/integrations/Axeptio/utils.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { isDefinedAndNotNull } from '../../utils/utils';
+import { isDefinedAndNotNull } from '../../utils/commonUtils';
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/integrations/Axeptio/utils.js:1]: It's available in `src/utils/commonUtils` Ln 11.

**Review by MoumitaM — DISMISSED** [review]

File: src/integrations/Engage/browser.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import logger from '../../utils/logUtil';
 import { NAME } from './constants';
 import { LOAD_ORIGIN } from '../../utils/ScriptLoader';
 import { refinePayload, getDestinationExternalID } from './utils.js';
-import { getDefinedTraits } from '../../utils/utils';
+import { getDefinedTraits } from '../../utils/commonUtils';
 import { removeUndefinedAndNullValues } from '../../utils/commonUtils';
```
Comment (MoumitaM) [inline src/integrations/Engage/browser.js:7]: We can combine these two lines.

Comment (saikumarrs) [reply src/integrations/Engage/browser.js:7]: There are more such issues even in files not part of the review. We need to fix all the linting issues at once.

**Comment by saikumarrs** [comment]
De-prioritizing it for now.

**Pull request closed without merging**